core/src/Kokkos_CopyViews.hpp

```diff
@@ -3560,8 +3560,7 @@ create_mirror_view_and_copy(
         Impl::MirrorViewType<Space, T, P...>::is_same_memspace>::type* =
         nullptr) {
   (void)name;
-  // Inherit fencing behavior of deep_copy
-  deep_copy(src, src);
+  fence();  // same behavior as deep_copy(src, src)
```

Comment:
```suggestion
  Kokkos::fence();  // same behavior as deep_copy(src, src)
```
since this is what we do everywhere else. Is it necessary to fence all execution spaces here or do you want `space.fence()` here?

Reply:
`Space` can be a memory space